client/src/CartContextProvider.jsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { useStickyState } from './utils/local-storage';
+
+export const CartContext = React.createContext([{}, () => {}]);
+
+export const CartContextProvider = ({ children }) => {
+  const [cartState, setCartState] = useStickyState({ ids: [] }, 'bonsai-cart');
```

Comment:
I'm honestly not sure about the efficiency/safety of connecting local state and context like this